UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentSalesCatalogueServiceTest.cs

```diff
@@ -0,0 +1,83 @@
+
+namespace UKHO.ExchangeSetService.Webjob.UnitTests.Services
+{
+    [TestFixture]
+    public class FulfilmentSalesCatalogueServiceTest
+    {
+        public ISalesCatalogueService fakeSalesCatalogueService;
+        public FulfilmentSalesCatalogueService fulfilmentSalesCatalogueService;
+        public string fakeBatchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272";
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeSalesCatalogueService = A.Fake<ISalesCatalogueService>();
+
+            fulfilmentSalesCatalogueService = new FulfilmentSalesCatalogueService(fakeSalesCatalogueService);
+        }
+
+        #region GetSalesCatalogueDataResponse
+        private SalesCatalogueDataResponse GetSalesCatalogueDataResponse()
+        {
+            return new SalesCatalogueDataResponse
+            {
+                ResponseCode = HttpStatusCode.OK,
+                ResponseBody = new List<SalesCatalogueDataProductResponse>()
+                {
+                    new SalesCatalogueDataProductResponse
+                    {
+                    ProductName="10000002",
+                    LatestUpdateNumber=5,
+                    FileSize=600,
+                    CellLimitSouthernmostLatitude=24,
+                    CellLimitWesternmostLatitude=119,
+                    CellLimitNorthernmostLatitude=25,
+                    CellLimitEasternmostLatitude=120
+                    }
+                }
+            };
+        }
+        #endregion
+
+        #region GetSalesCatalogueDataBadrequestResponse
+        private SalesCatalogueDataResponse GetSalesCatalogueDataBadrequestResponse()
+        {
+            return new SalesCatalogueDataResponse
+            {
+                ResponseCode = HttpStatusCode.BadRequest,
+                ResponseBody = null
+            };
+        }
+        #endregion
+
+        [Test]
+        public async Task WhenRequestGetSalesCatalogueDataResponse_ThenReturnsBadrequest()
+        {
+            A.CallTo(() => fakeSalesCatalogueService.GetSalesCatalogueDataResponse(A<string>.Ignored, A<string>.Ignored)).Returns(GetSalesCatalogueDataBadrequestResponse());
+
+            var response = await fulfilmentSalesCatalogueService.GetSalesCatalogueDataResponse(fakeBatchId, null);
+
+            Assert.AreEqual(HttpStatusCode.BadRequest, response.ResponseCode);
+        }
+
+        [Test]
+        public async Task WhenRequestGetSalesCatalogueDataResponse_ThenReturnsDataInResponse()
+        {
+            A.CallTo(() => fakeSalesCatalogueService.GetSalesCatalogueDataResponse(A<string>.Ignored, A<string>.Ignored)).Returns(GetSalesCatalogueDataResponse());
+
+            var response = await fulfilmentSalesCatalogueService.GetSalesCatalogueDataResponse(fakeBatchId, null);
+
+            Assert.IsNotNull(response);
+            Assert.AreEqual(HttpStatusCode.OK, response.ResponseCode);
+            Assert.IsInstanceOf(typeof(SalesCatalogueDataResponse), response);
+        }
```

Comment:
I think we only really need a single test as the method under test only really has one responsibility.

GetSalesCatalogueDataResponse is responsible for returning the result from salesCatalogueService.GetSalesCatalogueDataResponse.

The second Test could be called GetSalesCatalogueDataResponse_CallsSalesCatalogueServiceGetSalesCatalogueDataResponse.

The first test could be removed.

Reply:
As per suggestion i have done changes.